frontend/src/metabase/dashboard/components/ClickBehaviorSidebar/LinkOptions/LinkedEntityPicker.tsx

```diff
@@ -157,6 +178,61 @@ function LinkedEntityPicker({
     });
   }, [clickBehavior, updateSettings]);
 
+  const { data: dashboard } = useDashboardQuery({
+    enabled: isDashboard,
+    id: targetId,
+  });
+  const dashboardTabs = dashboard?.tabs ?? NO_DASHBOARD_TABS;
+  const defaultDashboardTabId: number | undefined = dashboardTabs[0]?.id;
+  const dashboardTabId = isDashboard
+    ? clickBehavior.tabId ?? defaultDashboardTabId
+    : undefined;
+  const dashboardTabExists = dashboardTabs.some(
+    tab => tab.id === dashboardTabId,
+  );
+  const dashboardTabIdValue =
+    typeof dashboardTabId === "undefined" ? undefined : String(dashboardTabId);
+
+  const handleDashboardTabChange = (value: string) => {
+    if (!isDashboard) {
+      throw new Error("This should never happen");
+    }
+
+    updateSettings({ ...clickBehavior, tabId: Number(value) });
+  };
+
+  useEffect(
+    function migrateUndefinedDashboardTabId() {
```

Comment:
Data migration on the FE is a known anti-pattern. If we need to it, this should be done on the BE.

Reply:
The entire [`visualization_settings.click_behavior`](https://github.com/metabase/metabase/blob/8fb7a7842528197567c523ae921b79ce055d3bf9/frontend/src/metabase-types/api/actions.ts#L181) logic is currently handled frontend-side, see [Slack discussion](https://metaboat.slack.com/archives/C057T1QTB3L/p1697633134404609).
I'm all for changing this but it would require a significant backend + frontend effort.